pkg/cli/webhook/gitlab.go

```diff
@@ -87,15 +87,19 @@ func (gl *gitLabConfig) askGLWebhookConfig(repoURL, controllerURL, apiURL string
 		}
 	}
 
-	RandomCrypto, randErr := rand.Prime(rand.Reader, 16)
-	if randErr != nil {
-		return randErr
-	}
-	data, marshalErr := json.Marshal(RandomCrypto)
-	if marshalErr != nil {
-		return marshalErr
+	if webhookSecret == "" {
+		RandomCrypto, randErr := rand.Prime(rand.Reader, 16)
```

Comment:
can we get that function to its own? (not sure why it was copy pasted around at first in the providers)

Reply:
sure, makes sense